### DISCUSSION TITLE
HELP NEEDED: How do Notes get their flags?

### DISCUSSION THREAD

**Discussion by Matthew-w56** [rism-digital/verovio#3618] [Q&A]
title.

Basically, I found a problem that causes the system to crash because a note is left with a stem, but with no flag. And it's a quarter note so it's expected to have a flag.  This causes an assert to fail.

It's part of the edit system and so I know where to add in any calls that would prompt the note to reset or generate it's appropriate flag, but I'm not quite sure how to do that.  I've had a hard time parsing the code that loads data to find how it happens there, so any expertise is welcome!  Basically, how do I grow a flag on a stem?  What kind of reset or init method would allow this to happen?

Thanks for any responses!

**Comment by lpugin** [comment]
Notes get their flags automatically based on their duration and when not part of a beam. As mentioned before, the editor is a test code that remains as a proof-of-concept and needs to be refactored.

**Reply by Matthew-w56** [reply]
Thanks for the reply.  I realize that notes usually get their flags automatically, and also that the edit code is not complete.  What I don't know is if there is any way to force the note to re-analyze it's flag needs, similar to my fix for the beams themselves by clearing their coords.

**Reply by Matthew-w56** [reply]
So any insight into how this might be done is what I'm looking for.  Thanks in advance with any responses
